### DISCUSSION TITLE
Getting Positional Information

### DISCUSSION THREAD

**Discussion by hay-kot** [pelletier/go-toml#832] [Questions]
Hello!

I'm building a rudimentary frontmatter linter for markdown and am using this library to support TOML. I am trying to provide additional context for users by giving the Line/Col numbers of the error. Looking through the issues I can see that this behavior was removed in v1

- https://github.com/pelletier/go-toml/issues/412

Looking through the existing code, it looks like even the AST that was recently exposed under unstable sub package doesn't even track that information. 

I guess my two questions are:

1. Is it possible with the current API to obtain that information? - If so can you point me in the right direction to do this?
2. Otherwise, is that something you would be open to tracking in the AST? I'd be willing to work on a PR to add that data.

**Comment by pelletier** [comment]
Sorry for the late response! The current decode/unmarshal API does provide location information on errors. When an error occurs, the parser tries to emit a `toml.DecodeError` error, which has a `Position()` method that returns the row and column where the error occurred. See https://pkg.go.dev/github.com/pelletier/go-toml/v2#example-DecodeError for an example on how to use it.

The direct AST API does not provide this facility. It returns a `ParserError` (https://pkg.go.dev/github.com/pelletier/go-toml/v2@v2.0.6/unstable#ParserError), which provides some information about the error but not the location exactly.

I'm open to exposing it on the unstable API, even not just for errors (for any node). The easiest path is probably to extract the code that turns a computes the location from a `ParserError` and expose it as a method on `ParserError`:

https://github.com/pelletier/go-toml/blob/58a592bbf8f9fb134915b26ff12fd369bd3ca846/errors.go#LL91-L95

https://github.com/pelletier/go-toml/blob/58a592bbf8f9fb134915b26ff12fd369bd3ca846/errors.go#L238-L252

Happy to review a PR!
